macOS/debugger.cpp

```diff
@@ -259,11 +306,14 @@ void *Debugger::RemoteAllocateNear(uint64_t region_min,
   max_address = (region_min < size) ? 0 : region_min - size;
   ret_address = RemoteAllocateBefore(min_address, max_address, size, protection);
   if (ret_address != NULL) {
+    MakeEntryRemoteAddress((mach_vm_address_t)ret_address, size);
     return ret_address;
   }
 
   // if all else fails, try within
-  return RemoteAllocateAfter(region_min, region_max, size, protection);
+  ret_address = RemoteAllocateAfter(region_min, region_max, size, protection);
+  MakeEntryRemoteAddress((mach_vm_address_t)ret_address, size);
```

Comment:
Here you'll try to MakeEntryRemoteAddress even if ret_address  is NULL. This will probably fail and result in some confusing error messages (errors related to shared memory rather than being unable to allocate).

Reply:
In MakeSharedMemory ( the old name is MakeEntryRemoteAddress ), I have validated the address whether it is NULL or not